pkg/types/types.go

```diff
@@ -52,10 +54,9 @@ type UpkeepResult struct {
 type UpkeepState uint
```

Comment:
I would suggest to split this into two different states.

One is a state per upkeep ID (inFlight or not)
One is a state per key (upkeepID+block) (eligible or not)